templates/cdk.master.auth-webhook.py

```diff
@@ -159,62 +165,66 @@ def check_secrets(token_review):
     return False
 
 
-def check_aws_iam(token_review):
+async def check_aws_iam(token_review):
     '''Check the request with an AWS IAM authn server.'''
     app.logger.info('Checking AWS IAM')
 
     # URL comes from /root/cdk/aws-iam-webhook.yaml
-    url = '{{ aws_iam_endpoint }}'
-    app.logger.debug('Forwarding to: {}'.format(url))
+    app.logger.debug('Forwarding to: {}'.format(AWS_IAM_ENDPOINT))
 
-    return forward_request(token_review, url)
+    return await forward_request(token_review, AWS_IAM_ENDPOINT)
 
 
-def check_keystone(token_review):
+async def check_keystone(token_review):
     '''Check the request with a Keystone authn server.'''
     app.logger.info('Checking Keystone')
 
     # URL comes from /root/cdk/keystone/webhook.yaml
-    url = '{{ keystone_endpoint }}'
-    app.logger.debug('Forwarding to: {}'.format(url))
+    app.logger.debug('Forwarding to: {}'.format(KEYSTONE_ENDPOINT))
 
-    return forward_request(token_review, url)
+    return await forward_request(token_review, KEYSTONE_ENDPOINT)
 
 
-def check_custom(token_review):
+async def check_custom(token_review):
     '''Check the request with a user-specified authn server.'''
     app.logger.info('Checking Custom Endpoint')
 
     # User will set the URL in k8s-master config
-    url = '{{ custom_authn_endpoint }}'
-    app.logger.debug('Forwarding to: {}'.format(url))
+    app.logger.debug('Forwarding to: {}'.format(CUSTOM_AUTHN_ENDPOINT))
 
-    return forward_request(token_review, url)
+    return await forward_request(token_review, CUSTOM_AUTHN_ENDPOINT)
 
 
-def forward_request(json_req, url):
+async def forward_request(json_req, url):
     '''Forward a JSON TokenReview request to a url.
 
     Returns True if the request is authenticated; False if the response is
     either invalid or authn has been denied.
     '''
     timeout = 10
+    resp_text = ''
     try:
-        try:
-            r = requests.post(url, json=json_req, timeout=timeout)
-        except requests.exceptions.SSLError:
-            app.logger.debug('SSLError with server; skipping cert validation')
-            r = requests.post(url, json=json_req, verify=False, timeout=timeout)
+        async with aiohttp.ClientSession() as session:
+            try:
+                async with session.post(url, json=json_req, timeout=timeout) as resp:
+                    resp_text = await resp.text()
+            except aiohttp.ClientSSLError:
+                app.logger.debug('SSLError with server; skipping cert validation')
+                async with session.post(url,
+                                        json=json_req,
+                                        verify_ssl=True,
```

Comment:
we should probably expose some sort of config to let users decided if they even want to go down the invalid cert path.  regardless, we just had a `ClientSSLError`, so i think you mean `False` here.